cpp/src/lance/io/writer.cc

```diff
@@ -108,7 +108,28 @@ ::arrow::Status FileWriter::WritePrimitiveArray(const std::shared_ptr<format::Fi
                                                 const std::shared_ptr<::arrow::Array>& arr) {
   auto field_id = field->id();
   auto encoder = field->GetEncoder(destination_);
-  ARROW_ASSIGN_OR_RAISE(auto pos, encoder->Write(arr));
+  auto type = field->type();
+
+  ::arrow::Result<std::shared_ptr<::arrow::Array>> physical_arr;
+  switch (type->id()) {
+    case ::arrow::TimestampType::type_id:
```

Comment:
would it make sense to create methods like "lance::arrow::is_datetime64_type" and "lance::arrow::is_datetime32_type" or something like that?